core/server/data/migrations/versions/2.2/2-add-integrations-and-api-key-tables.js

```diff
@@ -0,0 +1,33 @@
+const commands = require('../../../schema').commands;
+const logging = require('../../../../lib/common/logging');
+
+const tables = ['integrations', 'api_keys'];
+const _private = {};
+
+_private.addOrRemoveTable = (type, table, options) => {
```

Comment:
You seem to like this functional style of programming and I like it too! 

Usually though I would go with the "data last" principle for fn arguments, taking configuration/options as the first arguments and the thing to be operated on last.

That would enable you to then use a fn sig like `(type, options) => (table) -> { /...` so in your `Promise.each` below you can just do `Promise.each(tables, _private.addOrRemoveTable(type, options))`

This isn't a change request tho so :shrimp: :relaxed:

Reply:
Maybe it's because I'm not used to seeing it but I find `Promise.each(tables, _private.addOrRemoveTable(type, options))` harder to quickly parse due to the `table` argument being "hidden".

However I do agree that argument order should have more consideration given to it so that it's easier to use functions in a functional style when appropriate. TBH as this was a one-time migration file I didn't give it a lot of thought 😄 The `addOrRemoveTable` function looks like a good candidate for a general util function though so it's a good point to raise.

Reply:
I tried the more functional style out locally but I found it ended up with harder to follow code, I'll leave as-is for now 😄

Reply:
sure :blush: